Packaging.Targets/ArchiveBuilder.cs

```diff
@@ -277,25 +311,58 @@ protected void AddFile(string entry, string relativePath, string prefix, List<Ar
                     md5hash = md5hasher.GetHashAndReset();
                 }
 
-                // Only support ELF32 and ELF64 colors; otherwise default to BLACK.
-                ArchiveEntryType entryType = this.GetArchiveEntryType(fileHeader);
-
-                var mode = LinuxFileMode.S_IROTH | LinuxFileMode.S_IRGRP | LinuxFileMode.S_IRUSR | LinuxFileMode.S_IFREG;
-
-                if (entryType == ArchiveEntryType.Executable32 || entryType == ArchiveEntryType.Executable64)
+                ArchiveEntryType entryType;
+                if (fileMetadata != null && fileMetadata.GetDocumentation())
                 {
-                    mode |= LinuxFileMode.S_IXOTH | LinuxFileMode.S_IXGRP | LinuxFileMode.S_IWUSR | LinuxFileMode.S_IXUSR;
+                    entryType = ArchiveEntryType.Doc;
                 }
+                else
+                {
+                    // check whether it's an ELF file
+                    entryType = this.GetArchiveEntryType(fileHeader);
 
-                // If a Linux path has been specified, use that one, else, use the default one based on the prefix
-                // + current file name.
-                string name = fileMetadata?.GetLinuxPath();
+                    // check whether this is a .NET assembly
+                    if (entryType == ArchiveEntryType.None && File.Exists(fileName))
+                    {
+                        try
+                        {
+                            var assemblyName = AssemblyLoadContext.GetAssemblyName(fileName);
```

Comment:
This try/catch sounds very expensive, and we can probably apply some more logic to determine whether a file is a .NET executable.

Reply:
Exceptions are worrisome if they're on the hot path for performance sensitive code that runs on a sustained basis, like inside an ASP.NET app. This here is done once for each file during packaging of an app, so I don't find that of concern. The alternative is to pull in PE file decoder code, which is far more code (I started with that approach and discarded it). We could safeguard the section with a simple probe for the 'MZ' magic word being the first two bytes of the file content which should catch most cases.

Reply:
> We could safeguard the section with a simple probe for the 'MZ' magic word being the first two bytes of the file content which should catch most cases.

That sounds like a good compromise.

Reply:
I'll add that.